tools/check_format.py

```diff
@@ -635,7 +729,18 @@ def checkFormatVisitor(arg, dir_name, names):
   # python lists are passed as references, this is used to collect the list of
   # async results (futures) from running checkFormat and passing them back to
   # the caller.
-  pool, result_list = arg
+  pool, result_list, owned_directories, error_messags = arg
+
+  # Sanity check CODEOWNERS.  This doesn't need to be done in a multi-threaded
+  # manner as it is a small and limited list.
+  source_prefix = './source/'
+  full_prefix = './source/extensions/'
+  # Check to see if this directory is a subdir under /source/extensions
+  # Also ignore top level directories under /source/extensions since we don't
+  # need owners for source/extensions/access_loggers etc, just the subdirectories.
+  if dir_name.startswith(full_prefix) and '/' in dir_name[len(full_prefix):]:
```

Comment:
I'm quite late to the review but it seems that this could become more readable and portable if you used `os.path` functions as done elsewhere in this file. Maybe something like:
```python
if os.path.dirname(dir_name) == "./source/extensions":
```